lib/ransack/predicate.rb

```diff
@@ -9,34 +9,26 @@ def names
         Ransack.predicates.keys
       end
 
-      def names_by_decreasing_length
-        names.sort { |a, b| b.length <=> a.length }
-      end
```

Comment:
ActiveAdmin relies on this method:
```ruby
NoMethodError: undefined method 'names_by_decreasing_length' for Ransack::Predicate:Class

active_admin/filters/humanized.rb in predicates at line 50

          end

          def predicates
            Ransack::Predicate.names_by_decreasing_length
          end

          def ransack_predicate_translation

active_admin/filters/humanized.rb in current_predicate at line 46

          end

          def current_predicate
            @current_predicate ||= predicates.detect { |p| @body.end_with?("_#{p}") }
          end

          def predicates

active_admin/filters/humanized.rb in ransack_predicate_translation at line 54

          end

          def ransack_predicate_translation
            I18n.t("ransack.predicates.#{current_predicate}")
          end

          def active_admin_predicate_translation

active_admin/filters/humanized.rb in body at line 17

          end

          def body
            predicate = ransack_predicate_translation

            if current_predicate.nil?
              predicate = @body.titleize

active_admin/filters/resource_extension.rb in block (6 levels) in search_status_section at line 170

                    else
                      active.filters.each do |filter|
                        li do
                          span filter.body
                          b filter.value
                        end
                      end
```

Reply:
It seems like ActiveAdmin was duplicating behavior that Ransack already provided, so it could be simplified to:
```ruby
def current_predicate
  Ransack::Predicate.detect_from_string @body
end

Reply:
Ah, https://github.com/activeadmin/activeadmin/pull/4951 replaced this code with `Ransack::Translate.predicate`